Improved context when exiting with error

superlinkx opened the pull request:
fix: added context to every exit()
chore: refactored common connectionTest() and newAzureClient() calls into its own reusable function
chore: fixed a few linter issues, including two unhandled error cases
chore: added break if currentTask errors and updated log message
chore: fixed several instances of bad json tags that were flagged by go vet

superlinkx committed:
fix: added context to every exit()
chore: fixed a few linter issues, including two unhandled error cases
note: may want to improve what happens in those cases

superlinkx committed:
Merge branch 'main' into exit-error-wrapping

ddlees commented on cmd/start.go:
```diff
@@ -275,3 +277,16 @@ func updateClient(ctx context.Context, bheUrl url.URL, bheClient *http.Client) e
 	}
 }
+
+func connectAndCreateClient() client.AzureClient {
```
Tasty refactor. Can we move this to `utils.go`?

superlinkx replied:
Yeah, that's easy

ddlees commented on cmd/start.go:
```diff
@@ -128,7 +124,10 @@ func start(ctx context.Context) {
-								startTask(ctx, *bheInstance, bheClient, currentTask.Id)
+								if err := startTask(ctx, *bheInstance, bheClient, currentTask.Id); err != nil {
+									log.Error(err, "failed to start task")
+								}
```
If we fail to start the task then we should probably unset currenttask and try again after the next heartbeat

superlinkx replied:
Yeah, I didn't know for sure what the best error state would be here, so just logged for now. More than happy to add this behavior while we're here

superlinkx committed:
chore: moved connectAndCreateClient to utils.go
chore: added break if currentTask errors and updated log message
chore: fixed several instances of bad json tags that were flagged by go vet
